src/core_plugins/kibana/server/lib/management/saved_objects/relationships.js

```diff
@@ -0,0 +1,222 @@
+function formatId(id) {
+  return id.split(':')[1];
+}
+
+async function findDashboardRelationships(id, size, callCluster, savedObjectsClient) {
+  const kibanaIndex = savedObjectsClient._index;
+  const dashboard = await savedObjectsClient.get('dashboard', id);
+  const visualizations = [];
+
+  // TODO: should we handle exceptions here or at the parent level?
+  const panelsJSON = JSON.parse(dashboard.attributes.panelsJSON);
+  if (panelsJSON) {
+    const visualizationIds = panelsJSON.map(panel => panel.id);
+    const visualizationResponse = await callCluster('mget', {
+      body: {
+        docs: visualizationIds.slice(0, size).map(id => ({
+          _index: kibanaIndex,
+          _type: 'doc',
+          _id: `visualization:${id}`,
+          _source: [`visualization.title`]
+        }))
+      }
+    });
+
+    visualizations.push(...visualizationResponse.docs.reduce((accum, doc) => {
+      if (doc.found) {
+        accum.push({
+          id: formatId(doc._id),
+          title: doc._source.visualization.title,
+        });
+      }
+      return accum;
+    }, []));
+  }
+
+  return { visualizations };
+}
+
+async function findVisualizationRelationships(id, size, callCluster, savedObjectsClient) {
+  const kibanaIndex = savedObjectsClient._index;
+  const allDashboardsResponse = await callCluster('search', {
+    index: kibanaIndex,
+    size: 10000,
```

Comment:
FYI, you use `1000` in `scan_all_types.js`. Longer term, we need to make this actually page instead of doing what's happening here.